include/tvm/meta_schedule/integration.h

```diff
@@ -86,14 +86,11 @@ class MetaScheduleContextNode : public runtime::Object {
    * \param target Target info
    * \param dispatched A list of low-level IRs that the high-level IR could potentially dispatch to.
    *                   NullOpt means the dispatch needs to be done in the context.
-   * \return There are different types of the output
-   *         1) NullOpt if there is no feedback hint
-   *         2) tir::PrimFunc if `mod` should be lowered to a PrimFunc
-   *         3) relay::Function if `mod` should be dispatched to BYOC workflow
-   *         4) IRModule for unified dispatch
+   * \return IRModule Currently we only have to return tir::PrimFunc, but we wrap it
+   *                  under IRModule for more general future use.
    */
-  virtual Optional<ObjectRef> Query(runtime::String task_name, IRModule mod, Target target,
-                                    Optional<Array<IRModule>> dispatched) = 0;
+  virtual IRModule Query(runtime::String task_name, IRModule mod, Target target,
```

Comment:
Thanks for refactoring this interface! On the other hand, the interface itself might be unnecessary any more after the refactoring :-) Perhaps we would love to refactor this completely some time after this PR is merged!

Reply:
yeah, based on the current use, I want to change the interface to be `tir::Schedule -> tir::Schedule`, and rename `Query` to `ApplyToSchedule` or something like that. 

When we do need `IRModule -> IRModule` interface later, I think it's better to introduce another function. Rather than one interface that does everything.

Reply:
As I commented above, for other use cases that may or may not happen in the future, I think it's better to introduce another interface dedicated for it. BYOC, as currently implemented today, doesn't go through TE Compiler.